src/iarc7_motion/iarc_tasks/go_to_roomba_task.py

```diff
@@ -0,0 +1,238 @@
+
+                if not (self._height_checker.above_min_maneuver_height(
+                            self._drone_odometry.pose.pose.position.z)):
+                    return (TaskAborted(msg='Drone is too low'),)
+                elif not (self._z_holder.check_z_error(
+                    self._drone_odometry.pose.pose.position.z)):
+                    return (TaskAborted(msg='Z error is too high'),)
+                elif not self._check_roomba_in_sight():
+                    return (TaskAborted(msg='The provided roomba is not in sight of quad'),)
+
+                try:
+                    roomba_transform = self._tf_buffer.lookup_transform(
+                                        'level_quad',
+                                        self._roomba_id,
+                                        rospy.Time.now(),
+                                        rospy.Duration(self._TRANSFORM_TIMEOUT))
+                except (tf2_ros.LookupException,
+                        tf2_ros.ConnectivityException,
+                        tf2_ros.ExtrapolationException) as ex:
+                    rospy.logerr('ObjectGoToTask: Exception when looking up transform')
+                    rospy.logerr(ex.message)
+                    return (TaskAborted(msg='Exception when looking up transform during go_to roomba'),)
+
+                # Creat point centered at drone's center
+                stamped_point = PointStamped()
+                stamped_point.point.x = 0
+                stamped_point.point.y = 0
+                stamped_point.point.z = 0
+
+                # returns point distances of roomba to center point of level quad
+                self._roomba_point = tf2_geometry_msgs.do_transform_point(
+                                                    stamped_point, roomba_transform)
+
+                """ This is for tracking
+                if not self._check_max_roomba_range():
+                    return (TaskAborted(msg='The provided roomba is not found or not close enough to the quad'),)
+                """
+
+                if self._check_max_ending_roomba_range():
+                    return (TaskDone(),)
+
+                roomba_x_velocity = self._roomba_odometry.twist.twist.linear.x
+                roomba_y_velocity = self._roomba_odometry.twist.twist.linear.y
+                roomba_velocity = math.sqrt(roomba_x_velocity**2 + roomba_y_velocity**2)
+
+                x_overshoot = roomba_x_velocity/roomba_velocity * self._overshoot
+                y_overshoot = roomba_y_velocity/roomba_velocity * self._overshoot
+
+                # p-controller
+                x_vel_target = ((self._roomba_point.point.x + x_overshoot)
+                                    * self._K_X + roomba_x_velocity)
+                y_vel_target = ((self._roomba_point.point.y + y_overshoot)
+                                    * self._K_Y + roomba_y_velocity)
+
+                z_vel_target = self._z_holder.get_height_hold_response(
```

Comment:
If using the `TranslateStopPlanner` just pass in `GO_TO_HEIGHT` as the desired z component and it should take care of holding that height. Am I wrong about this @aftersomemath  or @amiller27 ?

Reply:
@aftersomemath would know better.  One thing I will say is that I don't think `GO_TO_HEIGHT` should be the variable name, first of all locals shouldn't be all caps, and second I don't think it's a descriptive name.

Reply:
The translate stop planner will hold the desired z height or change it you request a height other than what you currently are at.